From ca0e2ccb9323129b69036094145afa872799ce53 Mon Sep 17 00:00:00 2001
From: Arindam Nath <arindam.nath@amd.com>
Date: Fri, 31 Jul 2015 11:51:22 +0530
Subject: [PATCH 1/4] gstomxvideoenc: use scaling default macro's

* use #define macros instead of hardcoded value for default
* if omx failed to set the request scaling parameter then use default scaling value

Signed-off-by: Brijesh Singh <brijeshkumar.singh@amd.com>
Signed-off-by: Arindam Nath <arindam.nath@amd.com>
---
 omx/gstomxvideoenc.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/omx/gstomxvideoenc.c b/omx/gstomxvideoenc.c
index f42ec13..1a0ecea 100644
--- a/omx/gstomxvideoenc.c
+++ b/omx/gstomxvideoenc.c
@@ -395,8 +395,8 @@ gst_omx_video_enc_open (GstVideoEncoder * encoder)
       }
     }
 
-    if (self->scaling_width != 0xffffffff ||
-        self->scaling_height != 0xffffffff) {
+    if (self->scaling_width != GST_OMX_VIDEO_ENC_SCALING_WIDTH_DEFAULT ||
+        self->scaling_height != GST_OMX_VIDEO_ENC_SCALING_HEIGHT_DEFAULT) {
       OMX_CONFIG_SCALEFACTORTYPE scale_factor;
       GST_OMX_INIT_STRUCT (&scale_factor);
       scale_factor.nPortIndex = self->enc_out_port->index;
@@ -406,9 +406,9 @@ gst_omx_video_enc_open (GstVideoEncoder * encoder)
 
       if (err == OMX_ErrorNone) {
 
-        if (self->scaling_width != 0xffffffff)
+        if (self->scaling_width != GST_OMX_VIDEO_ENC_SCALING_WIDTH_DEFAULT)
           scale_factor.xWidth = self->scaling_width;
-        if (self->scaling_height != 0xffffffff)
+        if (self->scaling_height != GST_OMX_VIDEO_ENC_SCALING_HEIGHT_DEFAULT)
           scale_factor.xHeight = self->scaling_height;
 
         err =
@@ -417,10 +417,14 @@ gst_omx_video_enc_open (GstVideoEncoder * encoder)
         if (err == OMX_ErrorUnsupportedIndex) {
           GST_WARNING_OBJECT (self,
               "Setting scale configuration not supported by the component");
+          self->scaling_width = GST_OMX_VIDEO_ENC_SCALING_WIDTH_DEFAULT;
+          self->scaling_height = GST_OMX_VIDEO_ENC_SCALING_HEIGHT_DEFAULT;
         } else if (err == OMX_ErrorUnsupportedSetting) {
           GST_WARNING_OBJECT (self,
               "Setting scale configuration %u %u not supported by the component",
               self->scaling_width, self->scaling_height);
+          self->scaling_width = GST_OMX_VIDEO_ENC_SCALING_WIDTH_DEFAULT;
+          self->scaling_height = GST_OMX_VIDEO_ENC_SCALING_HEIGHT_DEFAULT;
         } else if (err != OMX_ErrorNone) {
           GST_ERROR_OBJECT (self,
               "Failed to set scale configuration: %s (0x%08x)",
-- 
1.9.1

